crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -386,7 +387,7 @@ impl<T: Component> WorldQuery for &mut T {
 pub struct WriteFetch<T> {
     storage_type: StorageType,
     table_components: NonNull<T>,
-    table_ticks: *mut ComponentTicks,
+    table_ticks: *const UnsafeCell<ComponentTicks>,
```

Comment:
I can't do that without creating a reference to the wrong element. This pointer is being offset before reading, thus it represents a reference to the whole vec slice. I can't get rid of that `UnsafeCell` and cast that to `*mut` at the same time. The only correct way to do that right now requires a `&UnsafeCell<ComponentTicks>` to exist first. This might be changed once `UnsafeCell::raw_get` gets stabilised.

Reply:
I see.